src/client/actions/MessageReactionAdd.js

```diff
@@ -28,6 +28,10 @@ class MessageReactionAdd extends Action {
 
     // Verify reaction
     if (message.partial && !this.client.options.partials.includes(PartialTypes.REACTION)) return false;
+    const existing =
+      message.reactions.cache.has(data.emoji.id || data.emoji.name) &&
+      message.reactions.cache.get(data.emoji.id || data.emoji.name).users.cache.get(user.id);
+    if (existing) return { message, reaction: existing, user };
```

Comment:
`existing` is a `User` here, but later returned as `reaction`.

I suggest something simpler:
```suggestion
    const existing = message.reactions.cache.get(data.emoji.id || data.emoji.name)
    if (existing && existing.users.cache.has(user.id)) return { message, reaction: existing, user };
```
That way there is no `Map#has` followed by a `Map#get` and the reaction will be correctly returned.

Reply:
yup, you're right! Just committed your suggestion and fixed that missing semicolon 😉 
thank you!